### DISCUSSION TITLE
WSL Webserver error due to permission

### DISCUSSION THREAD

**Discussion by silverstone1903** [apache/airflow#23914] [Q&A]
Hi all,

First of all it's a novice question, I've never used airflow before. I'm using WSL on Windows Server 2019 (1809). I installed Airflow with PostgreSQL (Postgre runs as a Windows Service) and didn't have any problem. Also I created a `airflow_home` folder in WSL file system (Linux FS) and didn't get any problem. However I changed `airflow_home` location by mounting C disk via `sudo mount -t drvfs C: / mnt/c` (basically I moved Linux folder to Windows folder) and I started to get error in the below. I came across a [Stackoverflow comment](https://stackoverflow.com/a/69493748) about it which briefly says Windows file system has no group permissons like Linux FS and this causes chmod fail. So moving files back will be the solution but the thing is I'm not the only user of airflow. I moved files under C: just because other users can reach the folder and add ther own dags. Otherwise it would be in something weird path in Linux and they won't copy files directly (directly means; right click and copy paste) to Linux FS. First, is there any solution for group permission error? And secondly how would you suggest to use dags folder access by every other server users (I moved files back to Linux FS).

By the way there is another [discussion](https://github.com/apache/airflow/discussions/13628) exactly facing the same issue but there is no detail about mounting Windows FS.

```bash
user@Server:/mnt/c/airflowhome$ airflow webserver
  ____________       _____________
 ____    |__( )_________  __/__  /________      __
____  /| |_  /__  ___/_  /_ __  /_  __ \_ | /| / /
___  ___ |  / _  /   _  __/ _  / / /_/ /_ |/ |/ /
 _/_/  |_/_/  /_/    /_/    /_/  \____/____/|__/
[2022-05-25 14:17:06,850] {dagbag.py:500} INFO - Filling up the DagBag from /dev/null
[2022-05-25 14:17:08,873] {manager.py:512} WARNING - Refused to delete permission view, assoc with role exists DAG Runs.can_create User
Running the Gunicorn Server with:
Workers: 4 sync
Host: 0.0.0.0:8080
Timeout: 120
Logfiles: - -
Access Logformat:
=================================================================
[2022-05-25 14:17:22 +0300] [18388] [INFO] Starting gunicorn 20.1.0
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/user/.local/lib/python3.6/site-packages/gunicorn/__main__.py", line 7, in <module>
    run()
  File "/home/user/.local/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 67, in run
    WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()
  File "/home/user/.local/lib/python3.6/site-packages/gunicorn/app/base.py", line 231, in run
    super().run()
  File "/home/user/.local/lib/python3.6/site-packages/gunicorn/app/base.py", line 72, in run
    Arbiter(self).run()
  File "/home/user/.local/lib/python3.6/site-packages/gunicorn/arbiter.py", line 198, in run
    self.start()
  File "/home/user/.local/lib/python3.6/site-packages/gunicorn/arbiter.py", line 137, in start
    self.pidfile.create(self.pid)
  File "/home/user/.local/lib/python3.6/site-packages/gunicorn/pidfile.py", line 45, in create
    os.chmod(self.fname, 420)
PermissionError: [Errno 1] Operation not permitted: '/mnt/c/airflowhome/airflow-webserver.pid'
^C[2022-05-25 14:17:47,169] {webserver_command.py:433} INFO - Received signal: 2. Closing gunicorn.
user@Server:/mnt/c/airflowhome$
```

Thanks in advance!

**Comment by potiuk** [comment]
There is no workaround. You need to use POSIX compliant filesystems for Airlfow.

Airflow does not (yet) work on Windows, it uses Linux environment - and makes assumptions about filesystems it uses. The best I can advise is something similar to https://dev.to/miftahafina/accessing-wsl2-files-from-windows-file-explorer-308o and making shortcuts for the users to access WSL2 filesystems via file explorer. But since I am not really a Windows user, I have no idea how feasilble it will be.

**Comment by silverstone1903** [comment]
Thank you for your answer @potiuk. Unfortunately due to IT restrictions there is no chance to run it on Unix. Also I specified Windows Server version on purpose because it doesn't support WSL2 (WSL2 needs 1903+ version). IT doesn't upgrade Win version, don't let use Docker as well (Docker would be a nice solution). I need to find the best optimal solution against these kind of constraints unfortunately. One possible solution is on my mind; moving back files to Linux FS and creating a bat script which copies files using WSL from Windows FS to airflow dags folder.

**Reply by potiuk** [reply]
Maybe. But I am afraid you will get more issues with WSL rather than WSL2  - I guess it's a bit lost effort and you will anyhow drop it. But I have no too much knowledge about it - this is totally not suported in Airflow and the big problem you have is that if you are going this route, you are pretty much on your own whenever you find any problems and tapping into community help will always end with "this is not really supported, you are on your own".

**Reply by silverstone1903** [reply]
Yes I'm aware of that, it will be a lost effort and the way I want to use is not sustainable in a long term. This is why I asked to community if there is a hidden answer which I didn't come across :) Thank you for your replies, I'll also check the some other alternatives for orchestration.
